### PR TITLE
Flatten and clean up reference held suarez implementation

### PR DESCRIPTION
**Pull request by glwagner**
Just a little clean up. I also tried to figure out the minimal packages we need, but this proved harder than I thought. If I include only the packages that have `using`, I get a mysterious error `MethodError: needs_concrete_A`. When I implement the restrictions here: 

https://github.com/glwagner/ClimaHeldSuarez.jl/blob/183176aac09c4dded5448a72caa3c5e6965ced6f/clima_core_held_suarez/hybrid/Project.toml#L28-L31

Then I get an error from ClimaCore, which is probably a bug (ie a missing `compat` entry in `ClimaCore` `Project.toml`).

Probably worth looking into at some point but I didn't have time to isolate the issue now, so I've contented myself with a seemingly-overly-large `Project.toml` for now.

### CODE REVIEW

**Comment by glwagner** [comment]
@LenkaNovak we should add a way to plot the solution just to make sure it still makes sense...

File: clima_core_held_suarez/nvtx.jl [hunk 1]
```diff
@@ -1,21 +0,0 @@
-# Some useful nvtx macros from https://github.com/CliMA/ClimaAtmos.jl/blob/main/examples/hybrid/nvtx.jl
-using NVTX, Colors
```
Comment (navidcy) [inline clima_core_held_suarez/nvtx.jl]: what's NVTX? probably should remove... but what about Colors?

Comment (glwagner) [reply clima_core_held_suarez/nvtx.jl]: I think both right? At least for now we don't use `Colors`

Comment (glwagner) [reply clima_core_held_suarez/nvtx.jl]: we should delete this whole file! (I don't know what it does)

**Comment by glwagner** [comment]
Uploading u.mp4…

**Comment by glwagner** [comment]
https://user-images.githubusercontent.com/15271942/231025713-6b8df019-8d33-45e9-96c8-e7671eae95f0.mp4

**Pull request merged**